src/middlewared/middlewared/plugins/smb.py

```diff
@@ -510,7 +510,7 @@ async def validate_smb(self, new, verrors):
             1. They contain invalid characters for NetBIOS protocol
             2. The name is identical to the NetBIOS workgroup.
             """
-            if not i:
+            if not new.get(i):
```

Comment:
What if for some reason netbiosname_b is null in the database and its not a truenas enterprise? Wont it still be an issue?